Exclude volatile DispVMs from Global Config

alimirjamali opened the pull request:
Excluding volatile DispVMs (disp*) while preserving named DispVMs (e.g. sys-usb & sys-firewall) where it is relevant specifically:

1. From General tab -> Clock qube ComboBox
2. From Updates tab -> dom0 update proxy ComboBox
3. While Saving settings (since volatile disposables have no prefs for saving)
4. From Whonix update proxy options
5. From USB Devices tab (as safety measure)

fixes: https://github.com/QubesOS/qubes-issues/issues/9924

alimirjamali commented:
> maybe there could be some wrapper/helper function to reduce the amount of boilerplate?

I amended the PR and implemented my original idea (a wrapper class around `qubesadmin.Qubes`).

Instead of many `try: except:`, it is done in two places.

alimirjamali commented:
> This does not fix the referenced issue, if the qube gets removed in the middle of global config startup, checking this property will fail too.

Ah. OK. In the above sentence, is this a reference to ordinary Qubes (e.g. AppVMs or DispVMs?). I believe it is logical for it to fail? Actually Global Config should indeed listen to domain add/remote events and show a big warning box and exit in this case. Why user wants to mess with the system while configuring it?

marmarta commented:
> Ah. OK. In the above sentence, is this a reference to ordinary Qubes (e.g. AppVMs or StandaloneVMs?). I believe it is logical for it to fail? Actually Global Config should indeed listen to domain add/remote events and show a big warning box and exit in this case. Why user wants to mess with the system while configuring it?

I think that mentally, a person might consider removing/adding qubes a similar kind of thing that configuring the system is. E.g. "I'll open the config to set an updatevm - oh I wanted to make a vpn qube - oh it's loading ..."

On the bigger issue, I think I feel like yeah, excluding volatile dispvms is a good idea here, but I guess I don't love the new property, I think it might be neater to overwrite Qubes.domains directly? But I might be wrong here.

alimirjamali commented:
> I think that mentally, a person might consider removing/adding qubes a similar kind of thing that configuring the system is. E.g. "I'll open the config to set an updatevm - oh I wanted to make a vpn qube - oh it's loading ..."

Ok. So in this case, for the original issue, I believe it has to be renamed from `Qubes global config fails to start if a DispVM shutdown while it's loading the settings` to something like `Qubes global config should be able to react to domains being added or removed while it is loading or running`. And I should change the PR comment from `fixes` to `related` (even though it should fix the original openQA problem).

> On the bigger issue, I think I feel like yeah, excluding volatile dispvms is a good idea here, but I guess I don't love the new property, I think it might be neater to overwrite Qubes.domains directly? But I might be wrong here.

The problem is that the `domains` is not a simple property. It is an entire `qubesadmin.app.VMCollection` class with many sub-methods and properties. Actually I tried to overwrite `domains` 1st but it became complex & problematic. But it will have the benefit that all changes will be in two classes and one source file (i.e. `global_config.py`). I will look at it again.

alimirjamali commented:
Ok. I rewrote the PR and added a specific unittest for a DispVM (`disp666`) which dies during Global Config running. The main modified source file is only one file (`global_config.py`). This should be good for review @marmarta
